src/makefiles/linux_cuda.mk

```diff
@@ -2,7 +2,8 @@
 CUDA_INCLUDE= -I$(CUDATKDIR)/include
 CUDA_FLAGS = -g -Xcompiler -fPIC --verbose --machine 32 -DHAVE_CUDA
 
-CXXFLAGS += -DHAVE_CUDA -I$(CUDATKDIR)/include 
-LDFLAGS += -L$(CUDATKDIR)/lib -Wl,-rpath=$(CUDATKDIR)/lib
-LDLIBS += -lcublas -lcudart #LDLIBS : The libs are loaded later than static libs in implicit rule
+CXXFLAGS := -DHAVE_CUDA -I$(CUDATKDIR)/include $(CXXFLAGS)
+LDFLAGS := -L$(CUDATKDIR)/lib -Wl,-rpath=$(CUDATKDIR)/lib $(LDFLAGS)
```

Comment:
Again, I don't see why it's necessary to repeat this pattern for all types of flags, to add things at the end rather than the beginning
Plus, the way you're doing it with LDLIBS could possibly make a difference to the linking order which might make the difference between compiling and not compiling for some setups.
Actually now I see how extensive the changes are that you want to make, I'm starting to have second throughts about this pull request.  All this stuff is hard to test and hard to debug when there are failures because we don't have access to all platforms.